crates/ra_syntax/src/syntax_node.rs

```diff
@@ -523,6 +523,10 @@ impl<'a> SyntaxElement<'a> {
         }
     }
 
+    pub fn as_ast_node<T: AstNode>(&self) -> Option<&T> {
```

Comment:
i'd rather do `it.as_node().and_then(T::Cast)` on the call-site. I am somehow not a fan of a lot of small helper methods, because the make call-site harder to understand (b/c you need to goto defenition to the helper method to understand what's going on).